cluster/gce/util.sh

```diff
@@ -319,16 +335,62 @@ function create-minion {
       --can-ip-forward \
       --metadata-from-file "$3"; then
         if (( attempt > 5 )); then
-          echo -e "${color_red}Failed to create instance $1 ${color_norm}"
+          echo -e "${color_red}Failed to create instance template $1 ${color_norm}"
           exit 2
         fi
-        echo -e "${color_yellow}Attempt $(($attempt+1)) failed to create node $1. Retrying.${color_norm}"
+        echo -e "${color_yellow}Attempt $(($attempt+1)) failed to create instance template $1. Retrying.${color_norm}"
         attempt=$(($attempt+1))
-        # Attempt to delete the disk for this node (the disk may have been created even
-        # if the instance creation failed).
-        gcloud compute disks delete "$1" --project "${PROJECT}" --zone "${ZONE}" --quiet || true
      else
-       break
+        break
+    fi
+  done
+}
+
+# Robustly try to create a managed instance group.
+# $1: The name of the instance group.
+# $2: The name of the instance template to use.
+# $3: The initial number of instances to create.
+function create-managed-instance-group {
+  detect-project
+  local attempt=0
+  while true; do
+    if ! gcloud preview managed-instance-groups --zone "${ZONE}" create "$1" \
+      --project "${PROJECT}" \
+      --base-instance-name "${NODE_INSTANCE_PREFIX}" \
+      --size $3 \
```

Comment:
Please use double quotes around `"$3"`, it's supposed to expand to a single token.